repo/fsrepo/fsrepo.go

```diff
@@ -34,55 +31,74 @@ var (
 
 	// packageLock must be held to while performing any operation that modifies an
 	// FSRepo's state field. This includes Init, Open, Close, and Remove.
-	packageLock sync.Mutex // protects openersCounter and lockfiles
-	// lockfiles holds references to the Closers that ensure that repos are
-	// only accessed by one process at a time.
-	lockfiles map[string]io.Closer
-	// openersCounter prevents the fsrepo from being removed while there exist open
-	// FSRepo handles. It also ensures that the Init is atomic.
-	//
-	// packageLock also protects numOpenedRepos
-	//
-	// If an operation is used when repo is Open and the operation does not
-	// change the repo's state, the package lock does not need to be acquired.
-	openersCounter *counter.Openers
-
-	// protects dsOpenersCounter and datastores
-	dsLock           sync.Mutex
-	dsOpenersCounter *counter.Openers
-	datastores       map[string]ds2.ThreadSafeDatastoreCloser
+	packageLock sync.Mutex
```

Comment:
I'm not quite sure, and it's no longer not in my way, so I left it alone for now. I feel like it should go as a struct field if anything, but then there are a few non-method functions that use it.